docs/configuration.md

```diff
@@ -0,0 +1,237 @@
+
+A boolean indicating whether assets should compiled or not allowing in-browser debugging.
+
+
+## ElasticSearch configuration
+
+### ELASTICSEARCH_URL
+
+**default**: 'localhost:9200'`
+
+The elasticsearch server url used for search indexing.
+
+```python
+ELASTICSEARCH_URL = 'elasticserver:9200'
+```
+
+## Mongoengine/Flask-Mongoengine options
+
+### MONGODB_HOST
+
+**default**: `localhost`
+
+The mongodb hostname used by udata.
+
+### MONGODB_PORT
+
+**default**: `27017`
+
+The mongodb post used by udata.
+
+### MONGODB_DB
+
+**default**: `udata`
+
+The mongodb database used by udata. When testing, the test database will use the same name suffixed by `-test`
+
+See [the official Flask-MongoEngine documentation](https://flask-mongoengine.readthedocs.org/en/latest/)
+for more details.
+
+
+## Celery options
+
+By default, uData is configured to use Redis as Celery backend and a cutomized MongoDB scheduler.
+
+The defaults are:
+
+```python
+BROKER_URL = 'redis://localhost:6379'
+BROKER_TRANSPORT_OPTIONS = {
+    'fanout_prefix': True,
+    'fanout_patterns': True,
+}
+CELERY_RESULT_BACKEND = 'redis://localhost:6379'
+CELERY_ACCEPT_CONTENT = ['pickle', 'json']
+CELERYD_HIJACK_ROOT_LOGGER = False
+CELERYBEAT_SCHEDULER = 'udata.tasks.Scheduler'
+CELERY_MONGODB_SCHEDULER_COLLECTION = "schedules"
+```
+
+You can see the full list of Celery options in the
+[Celery official documentation](http://docs.celeryproject.org/en/latest/configuration.html).
+
+
+## Flask-Mail options
+
+You can see the full configuration option list in
+[the official Flask-Mail documentation](https://pythonhosted.org/flask-mail/).
+
+### MAIL_DEFAULT_SENDER
+
+**default**: `'webmaster@udata'`
+
+The default identity used for outgoing mails.
+
+
+## Flask-OAuthlib options
+
+uData is Oauthlib to provide OAuth2 on the API.
+The full option list is available in
+[the official Flask-OAuthlib documentation](https://flask-oauthlib.readthedocs.org/en/latest/oauth2.html#configuration)
+
+
+### OAUTH2_PROVIDER_TOKEN_EXPIRES_IN
+
+**default**: `30 * 24 * 60 * 60` (30 days)
+
+The OAuth2 token duration.
+
+
+### OAUTH2_PROVIDER_ERROR_ENDPOINT
+
+**default**: `'oauth-i18n.oauth_error'`
+
+The OAuth2 error page. Do not modify unless you know what you do.
+
+
+## Flask-Cache options
+
+uData use Flask-Cache to handle cache and use Redis by default.
+You can see the full options list in
+[the official Flask-Cache documentation](https://pythonhosted.org/Flask-Cache/)
+
+### CACHE_TYPE
+
+**default**: `'redis'`
+
+The cache type, which can be adjusted to your needs.
+
+
+### CACHE_KEY_PREFIX
+
+**default**: `'udata-cache'`
+
+A prefix used for cache keys to avoid conflict with other middleware.
+It also allows you use the same backend with different instances.
+
+
+## Flask-FS options
+
+uData use Flask-FS as storage abstraction.
+
+
+
+## Example configuration file
+
+Here a sample configuration file:
+
+```python
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
```

Comment:
Both lines are required if you want to have unicode chars in `SITE_TITLE` for example.